modules/userdirectory-brightspace/src/main/java/org/opencastproject/userdirectory/brightspace/BrightspaceUserProviderInstance.java

```diff
@@ -0,0 +1,274 @@
+            .build(new CacheLoader<String, Object>() {
+              @Override
+              public Object load(String username) {
+                User user = loadUserFromBrightspace(username);
+                return user == null ? nullToken : user;
+              }
+            });
+
+    this.registerMBean(pid);
+  }
+
+  @Override
+  public float getCacheHitRatio() {
+    if (loadUserRequests.get() == 0)
+      return 0;
+    return (float) (loadUserRequests.get() - brightspaceWebServiceRequests.get()) / loadUserRequests.get();
+  }
+
+  private void registerMBean(String pid) {
+    this.loadUserRequests = new AtomicLong();
+    this.brightspaceWebServiceRequests = new AtomicLong();
+
+    try {
+      ObjectName name = BrightspaceUserProviderFactory.getObjectName(pid);
+      MBeanServer mbs = ManagementFactory.getPlatformMBeanServer();
+
+      try {
+        mbs.unregisterMBean(name);
+      } catch (InstanceNotFoundException ife) {
+        logger.debug("{} was not registered", name);
+      }
+
+      mbs.registerMBean(this, name);
+    } catch (Exception e) {
+      logger.error("Unable to register {} as an mbean: {}", this, e);
+    }
+
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#getName()
+   */
+  @Override
+  public String getName() {
+    return pid;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#getUsers()
+   */
+  @Override
+  public Iterator<User> getUsers() {
+    return Collections.emptyIterator();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#loadUser(java.lang.String)
+   */
+  @Override
+  public User loadUser(String userName) {
+    this.loadUserRequests.incrementAndGet();
+    logger.debug("getting user from cache");
+
+    try {
+      Object user = this.cache.getUnchecked(userName);
+      if (user != this.nullToken) {
+        logger.debug("Returning user {} from cache", userName);
+        return (User) user;
+      } else {
+        logger.debug("Returning null user from cache");
+        return null;
+      }
+    } catch (ExecutionError | UncheckedExecutionException ee) {
+      logger.warn("Exception while loading user {}", userName, ee);
+      return null;
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#countUsers()
+   */
+  @Override
+  public long countUsers() {
+    return 0L;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#getOrganization()
+   */
+  @Override
+  public String getOrganization() {
+    return this.organization.getId();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.security.api.UserProvider#findUsers(java.lang.String, int, int)
+   */
+  @Override
+  public Iterator<User> findUsers(String query, int offset, int limit) {
+    return Collections.emptyIterator();
```

Comment:
Same here. Why never return any user?

Reply:
I think it is because we don't need this method, not sure. From where would you use this method, is this to search for users from the UI?